internal/pkg/redis/client.go

```diff
@@ -145,6 +147,7 @@ func (c Client) Subscribe(topics []types.TopicChannel, messageErrors chan error)
 		return err
 	}
 
+	var previousErr error
```

Comment:
this variable should not be declared here, because it will cause the data race

Reply:
this should be declared in the for loop

Reply:
Fixed, but in testing I noticed DS Publish gets EOF error on first Publish after Redis is back up. Next publish works fine. Going to look at a Retry of EOF error.